lib/ews/soap/exchange_web_service.rb

```diff
@@ -271,11 +279,49 @@ def validate_version(exchange_version)
     end
 
     # Build the common elements in the SOAP message and yield to any custom elements.
-    def build_soap!(&block)
-      opts = { :server_version => server_version, :impersonation_type => impersonation_type, :impersonation_mail => impersonation_address }
+    def build_soap!(opts: {}, &block)
+      opts.merge!({ :server_version => server_version, :impersonation_type => impersonation_type, :impersonation_mail => impersonation_address })
       opts[:time_zone_context] = @time_zone_context if @time_zone_context
       EwsBuilder.new.build!(opts, &block)
     end
 
+    def build_get_user_settings_request_doc(builder:, users:, requested_settings:, opts: {})
```

Comment:
This method is a bit hacky, but I couldn't find an easier way of building those nodes. There's also a slight issue here with a namespace not getting generated, but I spent a few hours trying to figure that out without success and I'm not even sure it's gonna be required when making the request to exchange, so it might be worth waiting until we can actually do a real test.

Reply:
We could probably move these building header/nodes into `ews_builder.rb` as methods similar to #6